documenteer/sphinxconfig/stackconf.py

```diff
@@ -291,8 +292,111 @@ def _insert_graphviz_configs(c):
     return c
 
 
+def _insert_eups_version(c, eups_version=None):
+    """Insert information about the current EUPS tag into the configuration
+    namespace.
+
+    Parameters
+    ----------
+    eups_version
+        The EUPS version string (as opposed to tag). This comes from the
+        __version__ attribute of individual modules. It will be None for the
```

Comment:
I'm confused by this.  Doesn't the `__version__` string always get set by `scons` based on the `SHA1` of the git repository?  Why would it ever be None?

Reply:
Thanks @SimonKrughoff, this all because single package builds (with versions from sconsUtils) are to be handled different from full stack builds (on Jenkins where an EUPS_TAG variable exists). I rewrote the code to make this logic clearer and to also handle more types of EUPS tags (rc, etc).